discovery/stackit/server.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackit
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/stackitcloud/stackit-sdk-go/core/auth"
```

Comment:
> After [initial discussion in slack](https://cloud-native.slack.com/archives/C01AUBA4PFE/p1742031661146599), I tried to implement an own lightweight API client to avoid large increases on the binary size.

I still see the dependency here, do you mean you avoided a bigger dependency in? 

Nevertheless, do you mind roughly assessing the binary size after adding `github.com/stackitcloud/stackit-sdk-go/core` module?

Reply:
```
% git checkout main
% git rev-parse HEAD
bce72b93d91186785a675fcd45b5bca2669d30de
% make build > /dev/null 2>&1
% ls -l ./prometheus 
-rwxr-xr-x@ 1 jok  staff  153374178 Apr 11 10:39 ./prometheus
% git checkout stackit
Switched to branch 'stackit'
Your branch is up to date with 'origin/stackit'.
% git rev-parse HEAD
d7d384d835c12cf7eaa31e354751680e630c35c3
% make build > /dev/null 2>&1
% ls -l ./prometheus         
-rwxr-xr-x@ 1 jok  staff  153540082 Apr 11 10:44 ./prometheus
```

Calculating together here 
153540082 bytes / 153374178 bytes = 1.001 

results into a ~0.1% binary increase.

Reply:
LGTM, I think it's not significant increase. Thanks for the analysis!